### ISSUE TITLE
Smap needs memcpy on modern toolchains

### ISSUE THREAD

**Issue by AKuHAK** [ps2dev/ps2sdk#96]
It seems the modern gcc is optimising stuff on its own so smap needs memcpy to be included. Referring to this commit https://github.com/ps2dev/ps2sdk/commit/4c19331f7b90c8c4763a007a12d6b2fc5cf8fd57 it seems that it is safe to remove -Os flag, but i am not sure.
```cpp
mipsel-ps2-irx-gcc -D_IOP -fno-builtin -Os -G0 -Wall -I/work/iop/dev9/dev9/include -Isrc/ -Isrc/include -Iinclude/ -I/work/iop/kernel/include -I/work/common/include -mgpopt -G16384 -mno-check-zero-division -DLWIP_DHCP=1 -msoft-float -mno-explicit-relocs -o irx/smap.irx obj/main.o obj/smap.o obj/xfer.o obj/imports.o obj/exports.o -nostdlib -s  
/usr/local/ps2dev/iop/lib/gcc/mipsel-ps2-irx/8.2.0/../../../../mipsel-ps2-irx/bin/ld: obj/smap.o: in function `SMAPIoctl':
(.text+0x1a3c): undefined reference to `memcpy'
```

**Comment by sp193** [comment]
Os sets the optimization level. If you remove it, then the default is O0 - no optimization. Code compiled with no optimization is very, very long and slow, so you should use optimization. Os means to optimize for space.

As the IOP toolchain cannot automatically resolve dependencies, we have to generate an imports.lst file that suits the needs of the project (and the toolchain).

As there's no complete toolchain set for newer versions of Binutils+GCC, I don't know whether we should bother with addressing this additional dependency. But if you think we should anyway, You could perhaps add memcpy to imports.lst or to disable the specific optimization type that is resulting in the dependency on memcpy().

**Comment by AKuHAK** [comment]
Yes on my fork I just added memcpy into imorts list. I will try to search for gcc optimization flag... maybe it is safe to add this one function into imports list? Anyway no one cares about obsolete functions that are inside imports list. For example if we remove some function or dependency from module and forgot to remove it from imports list - function remain in its list forever and there is no easy way to check if function remain obsolete. When I was compiling ps2ident I removed almost all functions from list (just for fun) and ps2dent still compiles fun and without problems. Anyway this one function seems not very big - maybe it will be really safe to add it?
https://github.com/ps2homebrew/ps2sdk/commit/8e52bfdcc573043b02a3c70f057a6c02985657e9
